### PR TITLE
Pass PID to engines

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java [hunk 1]
```diff
@@ -61,34 +65,25 @@
...
     URL cFamilyPluginUrl = findEmbeddedCFamilyPlugin(getPluginsDir());
-    ConnectedGlobalConfiguration.Builder config = ConnectedGlobalConfiguration.builder()
-      .setLogOutput(globalLogOutput)
-      .setSonarLintUserHome(getSonarLintHome())
+    ConnectedGlobalConfiguration.Builder configBuilder = ConnectedGlobalConfiguration.builder()
       .addEnabledLanguages(enabledLanguages.toArray(new Language[0]))
-      .setExtraProperties(prepareExtraProps())
-      .setNodeJs(nodeJsManager.getNodeJsPath(), nodeJsManager.getNodeJsVersion())
-      .setWorkDir(getWorkDir())
-      .setModulesProvider(() -> modulesRegistry.getModulesForEngine(connectionId))
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java:84]: This got lost in the refactoring to `configureCommonEngine`, the management of modules is not the same for connected engines vs the standalone engine.

**Review by jblievremont — CHANGES_REQUESTED** [review]
A small difference on module management between connected and standalone engines got lost in the refactoring, else LGTM.

File: src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java:84]: Oups, good catch indeed!

**Review by jblievremont — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**